rllib/algorithms/algorithm_config.py

```diff
@@ -5741,6 +5744,119 @@ def num_cpus_for_local_worker(self, value):
         pass
 
 
+class DifferentiableAlgorithmConfig(AlgorithmConfig):
+    """An RLlib DifferentiableAlgorithmConfig builds a Meta algorithm from a given
+    configuration
+
+    .. testcode::
+
+        from ray.rllib.algorithm.algorithm_config import DifferentiableAlgorithmConfig
+        # Construct a generic config for an algorithm that needs differentiable Learners.
+        config = (
+            DifferentiableAlgorithmConfig()
+            .training(lr=3e-4)
+            .environment(env="CartPole-v1")
+            .learners(
+                differentiable_learner_configs=[
+                    DifferentiableLearnerConfig(
+                        DifferentiableTorchLearner,
+                        lr=1e-4,
+                    )
+                ]
+            )
+        )
+        # Similar to `AlgorithmConfig` the config using differentiable Learners can be
+        # used to build a respective `Algorithm`.
+        algo = config.build()
+
+
+    """
+
+    # A list of `DifferentiableLearnerConfig` instances that define differentiable
+    # `Learner`'s. Note, each of them needs to implement the `DifferentiableLearner`
+    # API.
+    differentiable_learner_configs: List[DifferentiableLearnerConfig]
+
+    def __init__(self, algo_class=None):
+        """Initializes the DifferentiableLearnerConfig instance.
+
+        Args:
+            algo_class: An optional Algorithm class that this config class belongs to.
+                Used (if provided) to build a respective Algorithm instance from this
+                config.
+        """
+        # Initialize the `AlgorithmConfig` first.
+        super().__init__(algo_class=algo_class)
+
+        # Initialize the list of differentiable learner configs to an empty list, which
+        # defines the default, i.e. the `MetaLearner` will have no nested updates.
+        self.differentiable_learner_configs: List[DifferentiableLearnerConfig] = []
+
+    def learners(
```

Comment:
Note that `learners` is also a superclass method, so we should accept **kwargs here and pass these to the super method first:
```
def learners(self, *, a, **kwargs):
    super().learners(**kwargs)
    
    self.a = a
    return self
```

Reply:
Yup, I ran over this in my meta learning algorithm implementation.